src/runtime_src/core/common/detail/linux/sysinfo.h

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+// Local - Include files
+#include "core/common/sysinfo.h"
```

Comment:
You should think of detail/linux/sysinfo.h being code that is pasted into core/common/sysinfo.cpp from where this header file is include.  The detail files are not stand-alone compilable, you should not have to include core/common/sysinfo.h from this file.
```suggestion
```